app/pages/SearchResultsPage/SearchResultsPage.tsx

```diff
@@ -34,22 +36,80 @@ interface SearchState extends ParsedQs {
 
 /** Wrapper component that handles state management, URL parsing, and external API requests. */
 export const SearchResultsPage = () => {
+  const [cookies] = useCookies(["googtrans"]);
+  const history = useHistory();
+  const { search } = useLocation();
   const { userLocation } = useAppContext();
   const [lastPush, setLastPush] = useState(Date.now());
-  const { search } = useLocation();
   const [expandList, setExpandList] = useState(false);
 
-  const searchState: SearchState = useMemo(
-    () => qs.parse(search.slice(1)),
-    [search]
-  );
+  const [searchState, setSearchState] = useState<SearchState | null>(null);
   const [searchRadius, setSearchRadius] = useState(
     searchState?.configure?.aroundRadius ?? "all"
   );
 
+  // In cases where we translate a query into English, we use this ref
+  // to represent the user's original, untranslated input. The untranslatedQuery
+  // is displayed in the UI and stored in the URL params.
+  // This untranslatedQuery value is also checked against when a new search is triggered
+  // to determine if the user has input a different query (vs. merely selecting refinements),
+  // in which case we need to call the translation API again
+  const untranslatedQuery = useRef<string | null | undefined>(null);
+  const translatedQuery = useRef(null);
```

Comment:
Just wondering, what's the rationale behind using refs instead of state for these variables? I'll have to admit that I haven't used refs much in React, and I think the only times I have ever used it were to get references to DOM elements.

Reply:
Updating refs does not trigger a state update. Since our code updates the `untranslatedQuery` and `translatedQuery` value within the useEffect hook, I was running into this React error: ```react-dom.development.js:23093 Uncaught Error: Maximum update depth exceeded. This can happen when a component repeatedly calls setState inside componentWillUpdate or componentDidUpdate. React limits the number of nested updates to prevent infinite loops.```

Reply:
(though this is now sort of moot as I've switched these to useState per your suggested changes to the useEffect block)